mlrun/api/utils/scheduler.py

```diff
@@ -367,8 +394,29 @@ async def submit_run_wrapper(scheduled_object, schedule_name):
         # otherwise all runs will have the same uid
         scheduled_object.get("task", {}).get("metadata", {}).pop("uid", None)
 
+        schedule_name_label = "schedule-name"
+        if "task" in scheduled_object and "metadata" in scheduled_object["task"]:
+            scheduled_object["task"]["metadata"].setdefault("labels", {})
+            scheduled_object["task"]["metadata"]["labels"][
+                schedule_name_label
+            ] = schedule_name
+
         db_session = create_session()
 
+        active_runs = get_db().list_runs(
+            db_session,
+            state=RunStates.running,
```

Comment:
You need to query everything that is not in `RunStates.terminal_states()`
IIRC we don't support it, so you'll need to implement this option in the runs DB level

Reply:
Also you're missing the project context here, remember there can be several schedules with the same name across different projects